tracer/src/Datadog.Trace/Debugger/Expressions/ProbeExpressionParser.cs

```diff
@@ -0,0 +1,483 @@
+// <copyright file="ProbeExpressionParser.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq.Expressions;
+using Datadog.Trace.Debugger.Models;
+using Datadog.Trace.Debugger.Snapshots;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+using static Datadog.Trace.Debugger.Expressions.ProbeExpressionParserHelper;
+
+namespace Datadog.Trace.Debugger.Expressions;
+
+internal partial class ProbeExpressionParser<T>
```

Comment:
It seems you are using the visitor pattern, do we want to communicate this in the naming of the class?